packages/taquito/src/contract/rpc-contract-provider.ts

```diff
@@ -731,6 +734,44 @@ export class RpcContractProvider
     );
   }
 
+  /**
+   * @description Creates a smart rollup originate operation
+   * @param SmartRollupOrigianteParams
+   * @returns An operation handle with results from the RPC node
+
+   */
+    async smartRollupOriginate(params: SmartRollupOriginateParams) {
+      const publicKeyHash = await this.signer.publicKeyHash();
+      const estimate = await this.estimate(
+        params,
+        this.estimator.smartRollupOriginate.bind(this.estimator)
+      );
+      const originationProof = await this.rpc.getOriginationProof({
+        kind: params.pvmKind,
+        kernel: params.kernel,
+      });
+      const completeParams = { ...params, originationProof };
+
+      const operation = await createSmartRollupOriginateOperation({
+        ...completeParams,
+        ...estimate,
+      });
+      const ops = await this.addRevealOperationIfNeeded(operation, publicKeyHash);
+      const prepared = await this.prepareOperation({ operation: ops, source: params.source });
+      const opBytes = await this.forge(prepared);
+      const { hash, context, forgedBytes, opResponse } = await this.signAndInject(opBytes);
+
+      return new SmartRollupOriginateOperation(
+        hash,
+        operation,
+        publicKeyHash,
+        forgedBytes,
+        opResponse,
+        context
+      );
+    }
+
+
```

Comment:
extraneous line break, please right click 'format document' on file to fix these